apollo-runtime/src/main/java/com/apollographql/apollo/ApolloClient.java

```diff
@@ -195,14 +192,10 @@ public Builder httpCache(@Nonnull ResponseCacheStore cacheStore, @Nonnull Evicti
     /**
      * Set the configuration to be used for normalized cache.
      *
-     * @param cacheStore       the cache store to use
-     * @param cacheKeyResolver the cache key resolver
      * @return The {@link Builder} object to be used for chaining method calls
      */
-    public Builder normalizedCache(@Nonnull CacheStore cacheStore,
-        @Nonnull CacheKeyResolver<Map<String, Object>> cacheKeyResolver) {
-      this.cache = new RealCache(checkNotNull(cacheStore, "cacheStore == null"),
-          checkNotNull(cacheKeyResolver, "cacheKeyResolver == null"));
+    public Builder normalizedCache(@Nonnull Cache cache) {
```

Comment:
I think in this case `RealCache` should be moved from the internal package. If we want to expose thread-safe `clear` we will have to allow the user to access the `RealCache` object. The only way to do that and keep the builder patter is to have user inject an instance of `RealCache`.

https://github.com/apollographql/apollo-android/issues/397 I plan to rename it to `ApolloStore` to better align with iOS and web naming.